Mathlib/Order/Incomparable.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Incomparability relation
+
+Two values in a preorder are incomparable whenever `¬ a ≤ b` and `¬ b ≤ a`.
+
+## Main declarations
+
+* `IncompRel`: The incomparability relation. `IncompRel r a b` means that `a` and `b` are related in
+  neither direction by `r`.
+-/
+
+open Function OrderDual
+
+variable {α : Type*} {a b c d : α}
+
+section Relation
+
+variable (r : α → α → Prop)
+
+/-- The incomparability relation `IncompRel r a b` means `¬ r a b` and `¬ r b a`. -/
+def IncompRel : α → α → Prop :=
+  AntisymmRel rᶜ
+
+theorem incompRel_iff {r} : IncompRel r a b ↔ ¬ r a b ∧ ¬ r b a := Iff.rfl
```

Comment:
I'd suggest swapping the definition with the one coming from the RHS of the lemma. This way:
* You don't need `incompRel_iff` as it's just the equational lemma now
* `simp [← antisymmRel_compl]` won't loop anymore